tasks/eva_2124/correct_assembly_for_study.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+import hashlib
+from argparse import ArgumentParser
+
+import pymongo
+from ebi_eva_common_pyutils.mongo_utils import get_mongo_connection_handle
+from tasks.eva_2124.load_synonyms import load_synonyms_for_assembly
+
+def get_SHA1(variant_rec):
+    """Calculate the SHA1 digest from the ref, study, contig, start, ref, and alt attributes of the variant"""
+    h = hashlib.sha1()
+    keys = ['seq', 'study', 'contig', 'start', 'ref', 'alt']
+    h.update('_'.join([str(variant_rec[key]) for key in keys]).encode())
+    return h.hexdigest().upper()
+
+
+def get_genbank(synonym_dictionaries, contig):
+    """
+    returns a tuple (genbank, was_already_genbank) or raises an exception if the contig was not found
+    """
+    by_name, by_assigned_molecule, by_genbank, by_refseq, by_ucsc = synonym_dictionaries
+    if contig in by_name:
+        return by_name[contig]['genbank'], False
+
+    if contig in by_assigned_molecule:
+        return by_assigned_molecule[contig]['genbank'], False
+
+    if contig in by_ucsc:
+        return by_ucsc[contig]['genbank'], False
+
+    if contig in by_refseq and by_refseq[contig]['is_genbank_refseq_identical']:
+        return by_refseq[contig]['genbank'], False
+
+    if contig in by_genbank:
+        return contig, True
+
+    raise Exception('could not find synonym for contig {}'.format(contig))
+
+
+def correct(mongo_user, mongo_password, mongo_host, studies, assembly_accession, chunk_size = 1000):
+    """
+    Connect to mongodb and retrieve all variants the should be updated, check their key and update them in bulk.
+    """
+    with get_mongo_connection_handle(
```

Comment:
I have an issue with this function because It force us to have a username and password. This mean you won't be able to test on EMBASSY.  
I changed it slightly wand will post it back when PR https://github.com/EBIvariation/eva-common-pyutils/pull/1 is merged.
Here is what I have: 
```python
def get_mongo_connection_handle(host, port=27017, username=None, password=None, authentication_database="admin") -> pymongo.MongoClient:
    mongo_connection_uri = "mongodb://"
    if username and password:
        mongo_connection_uri += '%s:%s@' % (username, quote_plus(password))
    mongo_connection_uri += '%s:%s/%s' % (host, port, authentication_database)
    return pymongo.MongoClient(mongo_connection_uri)
```